init/init.c

```diff
@@ -61,55 +65,170 @@ static void set_rlimits(const char *rlimits)
 }
 
 #ifdef SEV
-static char * get_luks_passphrase(int *pass_len)
+/*
+ * The LUKS passphrase is obtained from a KBS attestation server, complete an
+ * SNP attestation to get the passphrase.
+ */
+static char *get_luks_passphrase(int *pass_len)
 {
-	char *pass = NULL;
-	int len;
-	int fd;
-
-	pass = getenv("KRUN_PASS");
-	if (pass) {
-		*pass_len = strnlen(pass, MAX_PASS_SIZE);
-		return pass;
-	}
+        int fd, num_tokens, wid_found, url_found;
+        struct stat tc_stat;
+        char *pass, wid[256], url[256], tc_json[1024], *tok_start, *tok_end;
+        char full_path[512], *return_str;
+        jsmn_parser parser;
+        jsmntok_t *tokens;
+        size_t tok_size;
+        DIR *teeconfig;
+        struct dirent *tc_de;
+
+        return_str = NULL;
+
+        /*
+         * If a user registered the TEE config data disk with
+         * krun_set_data_disk(), it would appear as /dev/vdb in the guest.
+         * Mount this device and read the config.
+         */
+        if (mkdir("/dev", 0755) < 0 && errno != EEXIST) {
+                perror("mkdir(/dev)");
+                goto finish;
+        }
+
+        if (mount("devtmpfs", "/dev", "devtmpfs", MS_RELATIME, NULL) < 0 &&
+                        errno != EBUSY) {
+                perror("mount(devtmpfs)");
+
+                goto rmdir_dev;
+        }
+
+        if (mkdir("/teeconfig", 0755) < 0 && errno != EEXIST) {
+                perror("mkdir(/teeconfig)");
+
+                goto umount_dev;
+        }
+
+        if (mount("/dev/vdb", "/teeconfig", "ext4",
+                MS_NOEXEC | MS_NOSUID | MS_RELATIME, NULL) < 0) {
+                perror("mount(/dev/vdb)");
+
+                goto rmdir_teeconfig;
+        }
+
+        fd = open("/teeconfig/snp-config-attest.json", O_RDONLY);
```

Comment:
We're using `krun-sev.json` for the filename in `crun` and `oci2cw`, so it's probably a good idea using the same name here.

Reply:
Changed.